dask/bytes/core.py

```diff
@@ -149,6 +231,107 @@ def open_files(urlpath, compression=None, **kwargs):
                          compression=compression, **storage_options)
 
 
+def _expand_paths(path, name_function, num):
+    if isinstance(path, (str, unicode)):
+        if path.count('*') > 1:
+            raise ValueError("Output path spec must contain at most one '*'.")
+        if name_function is None:
+            name_function = build_name_function(num - 1)
+
+        if num > 1:
+            if not '*' in path:
+                path = os.path.join(path, '*.part')
+
+            formatted_names = [name_function(i) for i in range(num)]
+            if formatted_names != sorted(formatted_names):
+                warn("In order to preserve order between partitions "
+                     "name_function must preserve the order of its input")
+
+            paths = [path.replace('*', name_function(i))
+                     for i in range(num)]
+        else:
+            paths = [path]
+    elif isinstance(path, (tuple, list, set)):
+        assert len(path) == num
+        paths = path
+    else:
+        raise ValueError("""Path should be either"
+1.  A list of paths -- ['foo.json', 'bar.json', ...]
+2.  A directory -- 'foo/
+3.  A path with a * in it -- 'foo.*.json'""")
+    return paths
+
+
+def write_files(nfiles, urlpath, name_function=None, compression=None,
+                lazy=False, mode='wb', **kwargs):
+    """ Given path return dask.delayed file-like objects for writing
+
+    Parameters
+    ----------
+    values: list/set of values or collection
+        things that return bytes to be written into files
+    urlpath: string
+        Absolute or relative filepath, URL (may include protocols like
+        ``s3://``), or globstring pointing to data output.
+    name_function: function (None)
+        Way to convert each part into an identifier, if using a url globstring,
+        default: dask.utils.build_name_function
+    compression: string (None)
+        Compression to use.  See ``dask.bytes.compression.files`` for options.
+    lazy: bool (True)
+        whether to perform the write immediately, or return values which write
+        upon evaluation
+    **kwargs: dict
+        Extra options that make sense to a particular storage connection, e.g.
+        host, port, username, password, etc.
+
+    Examples
+    --------
+    >>> files = write_files('2015-*.csv')  # doctest: +SKIP
+    >>> files = open_files('s3://bucket/2015-*.csv.gz', compression='gzip')  # doctest: +SKIP
+
+    Returns
+    -------
+    List of ``dask.delayed`` objects that compute to file-like objects
+    """
+    if compression == 'infer':
+        if isinstance(urlpath, (tuple, list, set)):
+            compression = infer_compression(urlpath[0])
+        else:
+            compression = infer_compression(urlpath)
```

Comment:
Oh, I'm sorry, this is write_files rather than write_bytes.  I'm dumb.  My apologies.